contracts/atm/TLRToken.sol

```diff
@@ -230,24 +234,44 @@ contract TLRToken is
 
     /**
      *  @notice Withdrawl of tokens upon completion of vesting period
-     *  @return true if successful
      *
      */
     function withdrawVested() public whenNotPaused() isInitialized() {
         require(assignedTokens[msg.sender] > 0, "ACCOUNT_DOESNT_HAVE_VESTING");
 
         uint256 transferableTokens = _transferableTokens(msg.sender, block.timestamp);
-        approve(msg.sender, transferableTokens);
+        require(approve(msg.sender, transferableTokens), "APPROVAL_UNSUCCESSFUL");
+        approve(address(this), transferableTokens);
```

Comment:
Why are we approving itself to spend tokens?

Reply:
I think the line ```approve(address(this), transferableTokens);``` should be removed.

Another question: As I know this function is to withdraw funds. So, I am not understanding why we are calling the approve function before calling the transfer function (not transferFrom)?

Reply:
Removed as discussed